### PR TITLE
added function to get shell-by-shell confidences

### PR DESCRIPTION
**Pull request by mroczkowski**

### CODE REVIEW

File: deproject.py [hunk 1]
```diff
@@ -228,3 +228,43 @@ def get_density(self):
...
+        Run conf on each of the model parameters using the "onion-peeling" method:   
+
+         - First conf the outside shell model using the outer annulus spectrum
+         - Freeze the model parameters for the outside shell
+         - get confidences for the next inward shell / annulus and freeze those parameters
+         - Repeat until all datasets have been conf()-ed and all shell-by-shell error parameters determined.
+         - Return model parameters to original thawed/frozen status
+         - WARNING: This ignores the correlations between parameters
+
+        :rtype: None
```
Comment (taldcroft) [inline deproject.py:242]: Should be something like `:rtype: list of Sherpa get_conf_results() objects by shell`

**Comment by taldcroft** [comment]
Looks great, so now just two things:
- [ ] Add a `Confidence intervals` subsection under fitting, with basically the docstring text in your function as a prose explanation of what's happening.
- [ ] Add running conf to the M87 example.

File: deproject.py (hunk 1 above)
Comment (mroczkowski) [inline deproject.py:242]: Yes, that's true.  It returns a list, which I then used to get the
temperature and density error bars in the example I sent.

Cheers,
Tony

**Comment by mroczkowski** [comment]
I've added the second part (the M87 example now incorporates the error bars), but I'm not sure what the first is referring to.

**Comment by taldcroft** [comment]
I mean to add some new docs describing the conf functionality in this section:
   http://cxc.harvard.edu/contrib/deproject/#fitting
This is derived from the RST docs `docs/index.rst`

**Comment by DougBurke** [comment]
I've merged this, even though I plan to replace it with the functionality in #4

**Pull request merged**